src/jit/codegenxarch.cpp

```diff
@@ -5664,14 +5663,29 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
     // Determine return value size.
     emitAttr retSize = EA_PTRSIZE;
-    if (call->gtType == TYP_REF ||
-        call->gtType == TYP_ARRAY)
+#if defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+    emitAttr retSize1 = EA_UNKNOWN;
+    if (varTypeIsStruct(call->gtType))
     {
-        retSize = EA_GCREF;
+        // Make sure it is a multi-register passed struct.
```

Comment:
Please add to the comment "otherwise, for a struct passed in a single register, the call would have been retyped as a non-struct type."